spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -816,5 +816,47 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
   def getClassTagNumerics() : (Array[ClassTag[_]], Array[TensorNumeric[_]]) = {
     (Array(scala.reflect.classTag[T]), Array(ev))
   }
+
+  protected var optimMethod: OptimMethod[T] = null
+  protected var criterion: Criterion[T] = null
+  protected var vMethods: Array[ValidationMethod[T]] = null
+
+  /**
+   * Configures the learning process.
+   * Must call this before fit.
+   */
+  def compile(optimizer: OptimMethod[T], loss: Criterion[T],
+              metrics: Array[ValidationMethod[T]] = null): Unit = {
+    // TODO: shape inference
```

Comment:
Can we compile the model multiple times?

Reply:
Currently the behavior is that if compiled multiple times, the last compile configurations will take effect.